server/services/admin.ts

```diff
@@ -692,26 +692,27 @@ export = ({ strapi }: StrapiContext): IServiceAdmin => ({
     threadId: Id,
     body: ToBeFixed,
     author: StrapiAdminUser
-  ):Promise<Comment> {
+  ): Promise<Comment> {
     const entity = await strapi.db
       .query<Comment>(getModelUid("comment"))
       .findOne({
         where: {
           id: threadId,
         },
       });
-
+    
     return await strapi.db
       .query<Comment>(getModelUid("comment"))
       .create({
         data: {
           approvalStatus: "APPROVED",
           authorId: author.id,
-          authorName: author.username,
+          authorName: author.lastname ? `${author.firstname} ${author.lastname}` : author.username || author.firstname,
```

Comment:
What do you think about moving this logic to some kind of util function like `getAuthorName(author)`?